### DISCUSSION TITLE
Can't import third-party git repository: Invalid repository URL.

### DISCUSSION THREAD

**Discussion by 3ternal** [vercel/community#5322] [Help] [answered]
### Page to Investigate

https://vercel.com

### Steps to Reproduce

- Try to add any repository from https://gitgud.io (a gitlab instance)
- Vercel gives an error saying "Invalid repository URL" with no further information

**Comment by amyegan** [comment, marked as answer]
Hey @3ternal! You can [automate deployment from self-managed GitLab](https://vercel.com/guides/can-i-use-self-managed-gitlab-with-vercel) by using GitLab Pipelines and Vercel CLI. We have a guide that walks through setup with a starter script that works for most projects.

https://vercel.com/guides/how-can-i-use-gitlab-pipelines-with-vercel

**Reply by 3ternal** [reply]
Hey, thanks for the link, I'll look into that.

On https://vercel.com/new/git/third-party, it probably would make sense to add an error message saying "this git provider is not supported" or something like that. When I tried it, it was unclear whether the provider wasn't supported or whether there was a problem with the URL or the repo's privacy/access settings.
